### DISCUSSION TITLE
Is there a detailed introduction about cmdposition()?

### DISCUSSION THREAD

**Discussion by swarm5** [USC-ACTLab/crazyswarm#341] [Q&A]
Recently I want to use cmdposition(), but in the official documentation, I found that cmdposition() is not introduced much. I want to know more details about cmdposition.
1. After the cmdposition(pos) command is executed, what will be the status of cf? Will cf keep hovering? cf will fall directly? Still cf will hover for N seconds, after not receiving a new command, cf will fall.
2. What is the default flight speed for cmdposition(pos) command? Because after issuing the pos command, I have no way to estimate the time required to execute the command.

**Comment by jpreiss** [comment]
`cmdPosition` directly updates the setpoint for the position feedback controller.

1. The last one.
2. It depends on the gains of the feedback controller in use.

The argument to `cmdPosition` is not expected to be far away from the Crazyflie's current state. It sounds like `goTo` might be more appropriate for your needs.

**Comment by whoenig** [comment]
1. This is a streaming command, so if you don't send it for 500ms, the CF will go into emergency shutoff.
2. It changes the setpoint directly, so the duration depends on your controller. If you want a smooth trajectory, use go_to instead, which takes the desired time as input.

**Reply by jpreiss** [reply]
jinx!

**Comment by swarm5** [comment]
1. I have tested goto command many times, which is very easy to use for long-distance position movement!
2. However, under the PID controller, I use 20Hz to send goto command (the displacement of each segment is also very small, and the maximum flight speed is 0.5m/s). In this case, goto command will cause CF rollover, which seems to exceed the maximum acceleration of CF. Does Mellinger have better support for goto?
3. I want to input a curve so that CF can fly according to the trajectory of the curve. I'm going to send cmdposition() at 20Hz to approach the trajectory of the curve, but I haven't worked out a plan for goto at present.

**Reply by jpreiss** [reply]
sending `cmdPosition` with samples along your trajectory at 20Hz should work well, as long as your trajectory is within the Crazyflie's physical capabilities.
